R/pkg_ref_cache_downloads.R

```diff
@@ -1,13 +1,97 @@
-#' Cache a list of available help files as LaTeX objects
+#' Cache a list of package download metrics from a package reference
 #' @param n Number of days to look back with default value of 365 days
 #'
 #' @inheritParams pkg_ref_cache
 #' @family package reference cache
-#'
-#' @importFrom cranlogs cran_downloads
 #' @return a \code{pkg_ref} object
 #' @keywords internal
 #' @noRd
-pkg_ref_cache.downloads <- function(x, ..., n=365) {
-  cran_downloads(x$name, from=Sys.Date()-n, to=Sys.Date())
+pkg_ref_cache.downloads <- function(x, ..., n = 365) {
+  UseMethod("pkg_ref_cache.downloads")
+}
+
+#' @importFrom cranlogs cran_downloads
+#' @keywords internal
+pkg_ref_cache.downloads.pkg_ref <- function(x, ..., n = 365) {
+  downloads <- cran_downloads(x$name, from = Sys.Date() - n, to = Sys.Date())
```

Comment:
I don't think there is any need to dispatch a download method for `pkg_ref`, in fact we should return NA "by default" to account for sources that do not have download stats (e.g. source packages)